index.js

```diff
@@ -578,7 +569,7 @@ PersonaClient.prototype.updateProfile = function(guid, profile, token, callback)
     try {
         _.map([guid, token], function (arg) {
             if (!_.isString(arg)) {
-               throw "guid and token are required strings";
+                throw "guid and token are required strings";
```

Comment:
I recall telling @russellhill to remove try catch before and it caused problems because `map` is async. I don't want to figure out how to fix this given the time this work has already taken thus far, unless you have a working solution.